CMakeLists.txt

```diff
@@ -0,0 +1,397 @@
+cmake_minimum_required (VERSION 2.8)
+project (logcabin)
+set (LOGCABIN_VERSION_MAJOR 1)
+set (LOGCABIN_VERSION_MINOR 2)
+
+include_directories(BEFORE .)
+include_directories(BEFORE ./include)
+include_directories(BEFORE ./build)
+
+# c++ std should be defined in another way, but let make it work first
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-overflow -fPIC")
+
+set(CMAKE_PROTO_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+set(CMAKE_GTEST_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -DDEBUG")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_R} -O2 -DNDEBUG")
+
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    message("using clang")
+    set (FLAG_LIST 
+        -Wno-c++98-compat-pedantic 
+        -Wno-covered-switch-default 
+        -Wno-deprecated 
+        -Wno-disabled-macro-expansion 
+        -Wno-documentation-unknown-command 
+        -Wno-exit-time-destructors 
+        -Wno-float-equal 
+        -Wno-global-constructors 
+        -Wno-gnu-zero-variadic-macro-arguments 
+        -Wno-missing-noreturn 
+        -Wno-missing-prototypes 
+        -Wno-missing-variable-declarations 
+        -Wno-packed 
+        -Wno-padded 
+        -Wno-reserved-id-macro 
+        -Wno-shadow 
+        -Wno-shift-sign-overflow 
+        -Wno-switch-enum 
+        -Wno-undef 
+        -Wno-unknown-warning-option 
+        -Wno-unused-macros 
+        -Wno-unused-member-function 
+        -Wno-unused-parameter 
+        -Wno-used-but-marked-unused 
+        -Wno-vla 
+        -Wno-vla-extension 
+        -Wno-weak-vtables
+    )
+    foreach(FLAG in FLAG_LIST)
+        set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} ${FLAG}" ) 
+    endforeach()
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    message("using gcc")
+    set(FLAG_LIST 
+        -Wall 
+        -Wextra 
+        -Wcast-align 
+        -Wcast-qual 
+        -Wconversion 
+        -Weffc++ 
+        -Wformat=2 
+        -Wmissing-format-attribute 
+        -Wno-non-template-friend 
+        -Wno-unused-parameter 
+        -Woverloaded-virtual 
+        -Wwrite-strings 
+        -DSWIG)
+    foreach(FLAG ${FLAG_LIST})
+        set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} ${FLAG}" ) 
+    endforeach()
+# For some unknown reason, this suppresses some definitions
```

Comment:
This comment was attached to `-DSWIG` before; better to place the comment on `-DSWIG` and/or mention `-DSWIG` in the text.